src/cli/abi.ts

```diff
@@ -0,0 +1,198 @@
+import ts, { Decorator, NodeArray } from "typescript";
+import * as abi from "near-abi";
+import * as TJS from "typescript-json-schema";
+import { JSONSchema7 } from "json-schema";
+import * as fs from "fs";
+import { LIB_VERSION } from "../version.js";
+
+function parseMetadata(packageJsonPath: string): abi.AbiMetadata {
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+
+    let authors: string[] = [];
+    if (packageJson["author"]) authors.push(packageJson["author"]);
+    authors = authors.concat(packageJson["contributors"] || []);
+
+    return {
+        name: packageJson["name"],
+        version: packageJson["version"],
+        authors,
+        build: {
+            compiler: "tsc " + ts.version,
+            builder: "near-sdk-js " + LIB_VERSION
+        }
+    };
+}
+
+function getProgramFromFiles(
+    files: string[],
+    jsonCompilerOptions = {},
+    basePath = "./"
+): ts.Program {
+    // use built-in default options
+    const compilerOptions = ts.convertCompilerOptionsFromJson(jsonCompilerOptions, basePath).options;
+    const options: ts.CompilerOptions = {
+        noEmit: true,
+        emitDecoratorMetadata: true,
+        experimentalDecorators: true,
+        target: ts.ScriptTarget.ES5,
+        module: ts.ModuleKind.CommonJS,
+        allowUnusedLabels: true,
+    };
```

Comment:
So these overrides options load in basePath/tsconfig.json?

Reply:
So I tried to use contract's `tsconfig.json`, but then it picks up all TS files in this project, not just the one you directly specify (e.g. all examples under `/examples`, not just `counter-ts`). I will play around with this a bit more and see what can be done.

Reply:
I think I figured out the right way to deal with this, PTAL.

While we are at this, how do you think the path to `package.json` and `tsconfig.json` should be specified? I have just added a 3rd and 4th argument to the CLI command, but that's probably not what we want.